android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginViewModel.kt

```diff
@@ -1,33 +1,47 @@
 package vn.luongvo.kmm.survey.android.ui.screens.login
 
 import androidx.lifecycle.viewModelScope
-import kotlinx.coroutines.flow.catch
-import kotlinx.coroutines.launch
-import timber.log.Timber
+import kotlinx.coroutines.flow.*
 import vn.luongvo.kmm.survey.android.ui.base.BaseViewModel
-import vn.luongvo.kmm.survey.data.local.datasource.TokenLocalDataSource
+import vn.luongvo.kmm.survey.android.ui.navigation.AppDestination
+import vn.luongvo.kmm.survey.domain.usecase.IsLoggedInUseCase
 import vn.luongvo.kmm.survey.domain.usecase.LogInUseCase
 
 class LoginViewModel(
-    private val logInUseCase: LogInUseCase,
-    private val tokenLocalDataSource: TokenLocalDataSource
+    private val isLoggedInUseCase: IsLoggedInUseCase,
+    private val logInUseCase: LogInUseCase
 ) : BaseViewModel() {
 
-    fun logIn() {
-        // TODO sample request to verify network layer implementation
-        viewModelScope.launch {
-            logInUseCase(
-                email = "luong@nimblehq.co",
-                password = "12345678"
-            )
-                .catch { e ->
-                    Timber.e(e)
-                }
-                .collect {
-                    Timber.d("tokenType=${tokenLocalDataSource.tokenType}")
-                    Timber.d("accessToken=${tokenLocalDataSource.accessToken}")
-                    Timber.d("refreshToken=${tokenLocalDataSource.refreshToken}")
+    private val _isLoggedIn = MutableStateFlow<Boolean?>(null)
```

Comment:
@luongvo Can we remove the null value for the default?
```suggestion
    private val _isLoggedIn = MutableStateFlow<Boolean?>()
```

Reply:
@doannimble Nope, we can't. We need to have a null value while waiting for having a false or true value to show the Login page or navigate to the Home screen. 🤓

Reply:
@luongvo Thanks for verification, I just wonder if the default is null already without inputing any parameter 🤔